web/source/text/prediction/modelManager.ts

```diff
@@ -195,13 +195,26 @@ namespace com.keyman.text.prediction {
      * @param       {string}            event     event to handle
      * @param       {function(Event)}   func      event handler function
      * @return      {boolean}                     value returned by util.addEventListener
-     * Description  Wrapper function to add and identify KeymanWeb-specific event handlers
+     * Description  Wrapper function to add and identify handlers for ModelManager events
      */       
     ['addEventListener'](event: SupportedEventNames, func: SupportedEventHandler): boolean {
       let keyman = com.keyman.singleton;
       return keyman.util.addEventListener(ModelManager.EVENT_PREFIX + event, func);
     }
 
+    /**
+     * Function     removeEventListener
+     * Scope        Public
+     * @param       {string}            event     event to handle
+     * @param       {function(Event)}   func      event handler function
+     * @return      {boolean}                     value returned by util.addEventListener
+     * Description  Wrapper function to remove previously-added handlers for ModelManager events
+     */       
+    ['removeEventListener'](event: SupportedEventNames, func: SupportedEventHandler): boolean {
+      let keyman = com.keyman.singleton;
+      return keyman.util.removeEventListener(ModelManager.EVENT_PREFIX + event, func);
```

Comment:
1. Can't we remove the event listener w/o providing the func? 
2. We might as well remove event listeners in `setBanner` when we're toggling away from 'suggestion'

Reply:
1.  Please check kmwutils.ts, line 144.  In short, nope, we can't.
2.  Considering that `setBanner` always calls `_setBanner`, where this code resides, that's exactly what this does already.